--- linux-custom.old/drivers/video/fbdev/core/fbcon.c	2024-04-18 04:08:18.000000000 -0600
+++ linux-custom/drivers/video/fbdev/core/fbcon.c	2025-04-10 19:09:10.786595689 -0600
@@ -569,6 +569,7 @@ static void fbcon_prepare_logo(struct vc
 	if (fb_get_color_depth(&info->var, &info->fix) == 1)
 		erase &= ~0x400;
 	logo_height = fb_prepare_logo(info, ops->rotate);
+	logo_height+= (info->var.yres/2)-(logo_height/2);
 	logo_lines = DIV_ROUND_UP(logo_height, vc->vc_font.height);
 	q = (unsigned short *) (vc->vc_origin +
 				vc->vc_size_row * rows);
@@ -1004,8 +1005,8 @@ static void fbcon_init(struct vc_data *v
 
 	info = fbcon_info_from_console(vc->vc_num);
 
-	if (logo_shown < 0 && console_loglevel <= CONSOLE_LOGLEVEL_QUIET)
-		logo_shown = FBCON_LOGO_DONTSHOW;
+//	if (logo_shown < 0 && console_loglevel <= CONSOLE_LOGLEVEL_QUIET)
+//		logo_shown = FBCON_LOGO_DONTSHOW;
 
 	if (vc != svc || logo_shown == FBCON_LOGO_DONTSHOW ||
 	    (info->fix.type == FB_TYPE_TEXT))



--- linux-custom.old/drivers/video/fbdev/core/fbmem.c	2024-04-18 04:08:18.000000000 -0600
+++ linux-custom/drivers/video/fbdev/core/fbmem.c	2025-04-10 19:15:32.124961062 -0600
@@ -56,7 +56,7 @@ int min_dynamic_fb __read_mostly;
 
 bool fb_center_logo __read_mostly;
 
-int fb_logo_count __read_mostly = -1;
+int fb_logo_count __read_mostly = 1;
 
 struct fb_info *get_fb_info(unsigned int idx)
 {
@@ -499,6 +499,7 @@ static int fb_show_logo_line(struct fb_i
 		fb_set_logo(info, logo, logo_new, fb_logo.depth);
 	}
 
+	fb_center_logo = true;
 	if (fb_center_logo) {
 		int xres = info->var.xres;
 		int yres = info->var.yres;
